src/Utilities/Compiler/WellKnownTypeNames.cs

```diff
@@ -282,7 +282,8 @@ internal static class WellKnownTypeNames
         public const string SystemThreadingTasksGenericTask = "System.Threading.Tasks.Task`1";
         public const string SystemThreadingTasksGenericValueTask = "System.Threading.Tasks.ValueTask`1";
         public const string SystemThreadingTasksTask = "System.Threading.Tasks.Task";
-        public const string SystemThreadingTasksTaskCompletionSource = "System.Threading.Tasks.TaskCompletionSource`1";
+        public const string SystemThreadingTasksTaskCompletionSource = "System.Threading.Tasks.TaskCompletionSource";
+        public const string SystemThreadingTasksTaskCompletionSourceGeneric = "System.Threading.Tasks.TaskCompletionSource`1";
```

Comment:
Looking at the other definitions, I think it would be better to use `SystemThreadingTasksGenericTaskCompletionSource`.

Reply:
Renamed.